Update to 24w14potato

EnderKill98 opened the pull request:
Since I was porting some mods I can't live without, I thought it would be cool to have this as well.

The mod is not free of errors. I noticed this render artifact:

![grafik](https://github.com/squeek502/AppleSkin/assets/117233381/d52ba56d-6bc3-4f6c-9cad-1c8496758c52)

The hunerbar has also some transparency issue when indicating newly added food when holding an item and not fully fed:

![grafik](https://github.com/squeek502/AppleSkin/assets/117233381/6595823d-3c15-45e4-8584-9bebba0aff68)

Also, while I updated the CustomPayload to the new architecture, it could not test it yet (thought it starts which is already a good indicator).

This might be a useful stepping stone. Just verify that the server plugin integration works, fix the 2 render artifacts, and it should probably be 90% done for 1.20.5 as fabric calls the version "1.20.5-alpha.24.12.potato" in the `fabric.mod.json`.

I didn't test or try to port Forge. Only Fabric.

squeek502 commented:
Thanks!

Note that the `1.20.1-fabric` branch is not the latest 1.20.x Fabric branch, though--`1.20-fabric` is (sorry for the confusion). So, you might want to try starting from the 1.20-fabric branch instead since some of these changes might have already been taken care of.

Commit:
Update to 24w14potato

EnderKill98 commented:
Yeah I just noticed that. I thought some weird transparency issue was the cause for the black boxes, but the texture pointed to something like `minecraft/textures/gui/icons.png` instead of `appleskin/textures/icons.png` where they were.

Then the offsets were all wrong though, which is when I double checked and noticed the issue.

I force-pushed the branch, which is now based on `1.20-fabric`, as you mentioned. Rendering seems to work fine now.

Edit:
> since some of these changes might have already been taken care of

Yeah, the check for F3 and one mixin signature were already fixed. The custom payload stuff did change a lot though. I used one of minecrafts own Payloads as a guide. Fingers crossed, it just works :crossed_fingers: .

EnderKill98 commented:
Fun fact: You can't eat him, but he is still nutritious :rofl: 

![grafik](https://github.com/squeek502/AppleSkin/assets/117233381/bb1bccec-d1c9-4551-ae74-427deeeb17ce)

squeek502 committed:
Bump dependency versions

squeek502 committed:
Fix server-side networking of saturation/exhaustion syncing

Turns out that the payloads need to be registered server-side now for the server to be able to encode the packet.

Some other changes made:
- Made the Identifier for the exhaustion/saturation payloads match the latest Neoforge version
- `class` -> `record`
- A few style changes

squeek502 commented:
Great work on this! It turns out that the packets need to be registered server-side to avoid packet encoding failure when sending a sync packet from the server to the client, but other than that everything looks good.

Thanks!
